lib/src/http.dart

```diff
@@ -37,6 +44,20 @@ class _PubHttpClient extends http.BaseClient {
       : this._inner = inner == null ? new http.Client() : inner;
 
   Future<http.StreamedResponse> send(http.BaseRequest request) async {
+    if (_shouldAddMetadata(request)) {
+      request.headers["X-Pub-OS"] = Platform.operatingSystem;
```

Comment:
Do you mean the header names? I didn't realize you were expecting the implementation to use the exact names in the PRD. You're very much not supposed to use custom headers that don't begin with `X-`.